src/main/java/net/snowflake/client/core/StmtUtil.java

```diff
@@ -501,9 +501,20 @@ private static StmtOutput pollForOutput(String resultAsString,
       }
 
       if (pingPongResponseJson != null)
-      // raise server side error as an exception if any
+
       {
-        SnowflakeUtil.checkErrorAndThrowException(pingPongResponseJson);
+        // if JSON response is not successful, retry
+        if (!pingPongResponseJson.path("success").asBoolean())
+        {
+          retries++;
+        }
+        // raise server side error as an exception if retry count is exhausted
+        if (retries >= MAX_RETRIES)
```

Comment:
We want to raise an error if "success" entry is included in the response and the value is false.

Reply:
Do we want to retry several times if "success" + false value are the response? Which value is false in the JSON node?

Reply:
ok, here are the cases:
- If no "success" entry is included, we want to retry.
- If "success" entry is included, its value is `false` and no error `code` is attached, we want to retry.
- If "success" entry is included, its value is `false` and an error `code` is attached, we want to raise an error.

Reply:
Ah, makes sense! I separated the logic into a function.